src/app/views/home/home-view.tsx

```diff
@@ -310,37 +318,30 @@ export default function HomeView() {
   }, [location]);
 
   useEffect(() => {
```

Comment:
What I dont like in this useEffect is that it is too complicated. Do you really need to reattach the event listeners for fullscreen handling on each location change? Isn't it better to split up the Effect into two separate so that the event listeners are added only once, and then the tabs logic is changed on location? Also I dont think that we need all this state. There might be some state that can be derived but this is probably not the time for this.